main.py

```diff
@@ -1,16 +1,145 @@
-# This is a sample Python script.
+from contextlib import contextmanager
+from lit_class import BookModel, MagazineModel
+import csv
 
-# Press ⌃R to execute it or replace it with your code.
-# Press Double ⇧ to search everywhere for classes, files, tool windows, actions, and settings.
 
+def book_logger(func):
+    def wrapper(*args, **kwargs):
+        result = func(*args, **kwargs)
+        print("\n-LOG: The book has successfully been added to the library")
+        with open('log.txt', 'a') as file:
+            file.write(
 
-def print_hi(name):
-    # Use a breakpoint in the code line below to debug your script.
-    print(f'Hi, {name}')  # Press ⌘F8 to toggle the breakpoint.
+                f"Function '{func.__name__}' has been called.\n"
+                f"{result}",
+            )
+        return result
 
+    return wrapper
 
-# Press the green button in the gutter to run the script.
-if __name__ == '__main__':
-    print_hi('PyCharm')
 
-# See PyCharm help at https://www.jetbrains.com/help/pycharm/
+def book_checker(func):
+    def wrapper(*args, **kwargs):
+        found = False
+        for book in args[0].lit_list:
+            if book.lib_id == args[1]:
+                found = True
+        assert found, "There are no books in the library with this id"
+        result = func(*args, **kwargs)
+        return result
+
+    return wrapper
+
+
+class BookIterator:
+    def __init__(self, book_list: list):
+        self.books = book_list
+
+    def __iter__(self):
+        self.book_index = 0
+        return self
+
+    def __next__(self):
+        if self.book_index < len(self.books):
+            book = self.books[self.book_index]
+            self.book_index += 1
+            return book
+        raise StopIteration
+
+
+def author_books_gen(book_list: list, chosen_author: str):
+    for book in book_list:
+        if book.author == chosen_author:
+            yield book
+
+
+@contextmanager
+def file_opener(filename: str, mode: str):
+    """Opens the file in certain mode for further editing and closes it afterward"""
+    csvfile = open(filename, mode)
+    yield csvfile
+    csvfile.close()
+
+
+class Library:
+
+    def __init__(self, from_file: str = None):
+        self.lit_list = []
+        self.book_iterator = BookIterator(self.lit_list)
+
+    def show_books(self):
+        print("\nHere are the books, currently present at the library and their library id's: ", end=' ')
+        for book in self.book_iterator:
+            print("\"" + book.title + "\"" + " (" + str(book.lib_id) + ")", end=", ")
```

Comment:
Why you used fstring before, but here just concatenation? fstring is more readable and faster